linkerd/stack/metrics/src/service.rs

```diff
@@ -76,6 +91,87 @@ where
 
 impl<S> Drop for TrackService<S> {
     fn drop(&mut self) {
-        self.metrics.drop_total.incr();
+        if Arc::strong_count(&self._tracker) == 1 {
+            // If we're the last reference to the metrics, then we can
+            // increment the drop count.
+            self.metrics.drop_total.incr();
+        }
     }
 }
+
+#[cfg(test)]
+#[tokio::test(flavor = "current_thread")]
+async fn clone_drop() {
+    let metrics = Arc::new(Metrics::default());
+
+    let (a, _a) = tower_test::mock::pair::<(), ()>();
+    let a0 = TrackService::new(a, metrics.clone());
+    let a1 = a0.clone();
+
+    let (b, _b) = tower_test::mock::pair::<(), ()>();
+    let b0 = TrackService::new(b, metrics.clone());
+
+    drop(a1);
+    assert_eq!(metrics.drop_total.value(), 0.0, "Not dropped yet");
+
+    drop(b0);
+    assert_eq!(
+        metrics.drop_total.value(),
+        1.0,
+        "Dropping distinct service is counted"
+    );
+
+    drop(a0);
+    assert_eq!(
+        metrics.drop_total.value(),
+        2.0,
+        "Dropping last service clone counted"
+    );
+
+    assert_eq!(
+        metrics.create_total.value(),
+        0.0,
+        "No creates by the service"
+    );
+}
+
+#[cfg(test)]
+#[tokio::test(flavor = "current_thread", start_paused = true)]
+async fn clone_poll_ready() {
+    let metrics = Arc::new(Metrics::default());
+    let (mut svc0, mut handle) =
+        tower_test::mock::spawn_with::<(), (), _, _>(|svc| TrackService::new(svc, metrics.clone()));
+
+    handle.allow(0);
+    tokio_test::assert_pending!(svc0.poll_ready());
+    let mut svc1 = svc0.clone();
+    assert!(svc0.get_ref().blocked_since.is_some());
+    assert!(svc1.get_ref().blocked_since.is_none());
+
+    tokio_test::assert_pending!(svc1.poll_ready());
+    assert!(svc0.get_ref().blocked_since.is_some());
+    assert!(svc1.get_ref().blocked_since.is_some());
+
+    time::sleep(time::Duration::from_secs(1)).await;
+    handle.allow(2);
+    tokio_test::assert_ready_ok!(svc0.poll_ready());
+    tokio_test::assert_ready_ok!(svc1.poll_ready());
+    assert!(svc0.get_ref().blocked_since.is_none());
+    assert!(svc1.get_ref().blocked_since.is_none());
+
+    assert_eq!(
+        metrics.ready_total.value(),
+        2.0,
+        "Both clones should be counted discretely"
+    );
+    assert_eq!(
+        metrics.not_ready_total.value(),
+        2.0,
+        "Both clones should be counted discretely"
+    );
+    assert_eq!(
+        metrics.poll_millis.value(),
+        2000.0,
+        "Both clones should be counted discretely"
+    );
+}
```

Comment:
nit: since there's two tests here, maybe stick them in a module?
```suggestion
#[cfg(test)]
mod tests {
    use super::*;

    #[tokio::test(flavor = "current_thread")]
    async fn clone_drop() {
        let metrics = Arc::new(Metrics::default());

        let (a, _a) = tower_test::mock::pair::<(), ()>();
        let a0 = TrackService::new(a, metrics.clone());
        let a1 = a0.clone();

        let (b, _b) = tower_test::mock::pair::<(), ()>();
        let b0 = TrackService::new(b, metrics.clone());

        drop(a1);
        assert_eq!(metrics.drop_total.value(), 0.0, "Not dropped yet");

        drop(b0);
        assert_eq!(
            metrics.drop_total.value(),
            1.0,
            "Dropping distinct service is counted"
        );

        drop(a0);
        assert_eq!(
            metrics.drop_total.value(),
            2.0,
            "Dropping last service clone counted"
        );

        assert_eq!(
            metrics.create_total.value(),
            0.0,
            "No creates by the service"
        );
    }

    #[tokio::test(flavor = "current_thread", start_paused = true)]
    async fn clone_poll_ready() {
        let metrics = Arc::new(Metrics::default());
        let (mut svc0, mut handle) =
            tower_test::mock::spawn_with::<(), (), _, _>(|svc| TrackService::new(svc, metrics.clone()));

        handle.allow(0);
        tokio_test::assert_pending!(svc0.poll_ready());
        let mut svc1 = svc0.clone();
        assert!(svc0.get_ref().blocked_since.is_some());
        assert!(svc1.get_ref().blocked_since.is_none());

        tokio_test::assert_pending!(svc1.poll_ready());
        assert!(svc0.get_ref().blocked_since.is_some());
        assert!(svc1.get_ref().blocked_since.is_some());

        time::sleep(time::Duration::from_secs(1)).await;
        handle.allow(2);
        tokio_test::assert_ready_ok!(svc0.poll_ready());
        tokio_test::assert_ready_ok!(svc1.poll_ready());
        assert!(svc0.get_ref().blocked_since.is_none());
        assert!(svc1.get_ref().blocked_since.is_none());

        assert_eq!(
            metrics.ready_total.value(),
            2.0,
            "Both clones should be counted discretely"
        );
        assert_eq!(
            metrics.not_ready_total.value(),
            2.0,
            "Both clones should be counted discretely"
        );
        assert_eq!(
            metrics.poll_millis.value(),
            2000.0,
            "Both clones should be counted discretely"
        );
    }
}
```